annotations/Makefile

```diff
@@ -25,17 +25,26 @@ DIST_JAR = $(DIST_JAR_DIR)/$(JAR_NAME)
 
 CLASSES_DIR = $(BUILD_DIR)/classes
 JAVA_SOURCE_DIR = src/main/java
+JAVA9_CLASSES_DIR = $(BUILD_DIR)/classes-java9
+JPMS_CLASS = $(JAVA9_CLASSES_DIR)/module-info.class
 
 SOURCE_JAVA_FULL = $(ANNOTATION_SOURCE_JAVA:%=$(JAVA_SOURCE_DIR)/%)
+SOURCE_JAVA9_MODULE = $(ANNOTATION_MODULE_INFO:%=$(JAVA_SOURCE_DIR)/%)
 
-$(BUILD_DIR) $(CLASSES_DIR) $(DIST_JAR_DIR):
+$(BUILD_DIR) $(CLASSES_DIR) $(JAVA9_CLASSES_DIR) $(DIST_JAR_DIR):
 	@mkdir -p $@
 
-$(JAR): $(SOURCE_JAVA_FULL) | $(BUILD_DIR) $(CLASSES_DIR)
+$(JPMS_CLASS):
+	@echo Building j2objc annotations JPMS module
+	@$(JAVAC) -sourcepath $(JAVA_SOURCE_DIR) -encoding UTF-8 --release 9 -d $(JAVA9_CLASSES_DIR) \
+		-nowarn $(SOURCE_JAVA9_MODULE)
+	@rm -fr $(JAVA9_CLASSES_DIR)/com
+
+$(JAR): $(SOURCE_JAVA_FULL) | $(BUILD_DIR) $(CLASSES_DIR) $(JAVA9_CLASSES_DIR) $(JPMS_CLASS)
 	@echo Building j2objc annotations
 	@$(JAVAC) -encoding UTF-8 -d $(CLASSES_DIR) -source 1.8 -target 1.8 \
 	    -nowarn $^
-	@jar cf $(JAR) -C $(CLASSES_DIR) .
+	@jar cf $(JAR) -C $(CLASSES_DIR) . --release 9 -C $(JAVA9_CLASSES_DIR) .
```

Comment:
Main Makefile build logic for the MRJAR/Java9 change. JPMS module is built first so that `module-info.class` is available by the time `jar cf ...` runs, since `javac` and `jar` are in one block.

The main class root (`com/`) doesn't exist in `classes-java9` by the time this runs, so just the `module-info.class` is included in the Java 9 root. This results in the following JAR layout:

```
j2objc-annotations.jar/
├─ com/
│  ├─ google/
├─ META-INF/
│  ├─ versions/
│  │  ├─ 9/
│  │  │  ├─ module-info.class
│  ├─ MANIFEST.MF
```

Full JAR as built by Maven:
<img width="808" alt="Screenshot 2024-03-07 at 3 48 51 PM" src="https://github.com/google/j2objc/assets/171897/a44076d2-80f3-428c-b214-2de3e05565c3">

Full JAR as built by Makefile:
<img width="749" alt="Screenshot 2024-03-07 at 3 49 35 PM" src="https://github.com/google/j2objc/assets/171897/e01c5b8b-c4a0-4385-9429-ad615947617f">

Reply:
Thanks for the thorough explanation!